src/app/web-components/tabset/tabset.component.ts

```diff
@@ -43,13 +61,42 @@ export class TabsetComponent implements AfterViewInit {
             input.id = button.id;
             input.name = tabsetName;
             input.checked = !i;
+            this.tabs = [...this.tabs, input];
 
             tab.append(...content[i]);
             tabset.append(input, tab);
         });
+
+        if (this.subscription) {
+            this.subscription.unsubscribe();
+        }
```

Comment:
It looks like we don't need this part, it won't go into the 71nd line, since unsubscribing occurs when the component is destroyed, and this is our app component.